### ISSUE TITLE
Support Python 3.12 PEP 695 - Type Parameter Syntax

### ISSUE THREAD

**Issue by ericbn** [bwhmather/ssort#102]
### Is your feature request related to a problem? Please describe.

Currently ssort does not support Python 3.12 [PEP 695 - Type Parameter Syntax](https://peps.python.org/pep-0695/).

Simple example:

```python
from decimal import Decimal

type N = Decimal | float


def roundint(n: N) -> int:
    return int(round(n))
```

Currently running ssort with the above code fails with `NotImplementedError: AST traversal for 'TypeAlias' is not implemented`

### Describe the solution you'd like

I'd like ssort to support parsing Python 3.12 code that uses PEP 695 syntax. 

### Describe alternatives you've considered

An alternative could be implementing ssort in ruff, since it already supports parsing PEP 695 syntax. See https://github.com/astral-sh/ruff/issues/3946 and https://github.com/astral-sh/ruff/issues/5062

### Additional context

Using Python 3.12 and ssort revision 3bd130d.

**Comment by bwhmather** [comment]
Thank you for reporting this.  Volunteers always welcome for this sort of thing.  Should "just" be a case of providing implementations for `ssort._ast.iter_child_nodes`, `ssort._ast.get_bindings` and `ssort._requirements.get_requirements`.

I also support your suggestion of reimplementing in `ruff`, although it probably isn't the quickest way to get python 3.12 support.

**Comment by mwchase** [comment]
As far as PEP 695 goes, it should also handle [type parameter declarations](https://peps.python.org/pep-0695/#type-parameter-declarations). eg:
```python
class Example[T]:
    attr: T
```
Is rejected by `ssort 0.12.4` on Python 3.12, with the message:
```
ERROR: unresolved dependency 'T' in example.py: line 2, column 10
1 file was not sortable
```

**Comment by ericbn** [comment]
Opened PR #115 that solves the TypeAlias traversal in Python 3.12.

**Comment by ericbn** [comment]
This was fixed in release version [0.13.0](https://github.com/bwhmather/ssort/releases/tag/0.13.0)

**Comment by bwhmather** [comment]
Which has now, finally, made it to pypi.  Thank you @ericbn.
